docs/install.rst

```diff
@@ -130,18 +138,33 @@ then responsible for maintaining venv(s) with the correct RCT stack (matching th
 used by the client-side RCT stack), the `scalems` package, and any dependencies of the
 workflow.
 
-.. note:: Environment management for RP Tasks is under active development.
-
-    As of RP 1.6.7, a traditional :py:class:`~radical.pilot.Task` does not have explicitly Python-aware
-    environment preparation, though users are free to activate Task venvs using
-    :py:data:`~radical.pilot.TaskDescription.pre_exec`.
-    :py:mod:`radical.pilot.raptor`
-    `Workers <https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/raptor/worker.py>`__
-    have some of the RP stack injected into their environment, in addition to allowing *pre_exec*.
-
-    These details are subject to rapid evolution for the foreseeable future.
-
-    See also https://github.com/radical-cybertools/radical.pilot/pull/2312
+Task environment
+~~~~~~~~~~~~~~~~
+
+In addition to :py:attr:`~radical.pilot.TaskDescription.pre_exec`, TaskDescriptions have
+:py:attr:`~radical.pilot.TaskDescription.pre_launch`
+and :py:attr:`~radical.pilot.TaskDescription.pre_rank`
+hooks to set up the task environment.
+Note that, in addition to the attribute descriptions,
+there is additional discussion at the bottom of the
+:py:class:`~radical.pilot.TaskDescription` class documentation section.
+
+RADICAL Pilot now allows a :py:class:`~radical.pilot.Task` some explicitly Python-aware
+environment preparation, (though users are still free to activate Task venvs using
+:py:data:`~radical.pilot.TaskDescription.pre_exec`).
+TaskDescription may use :py:attr:`~radical.pilot.TaskDescription.named_env` to identify
+a virtual environment to be activated for the Task.
+The virtual environment may be an existing virtual environment, or a new environment,
+scheduled for creation with :py:func:`~radical.pilot.Pilot.prepare_env`.
+
+In addition to supporting :py:attr:`~radical.pilot.TaskDescription.named_env` and the
+other task environment hooks,
+:py:class:`~radical.pilot.raptor.Master` and :py:class:`~radical.pilot.raptor.Worker`
+tasks have some of the RP stack injected into their environment.
+Raptor tasks launched through a Worker are executed in new processes
```

Comment:
What happens if a raptor TaskDescription has `named_env`? Or `pre_exec`? What if these are different from those of the Worker that handles the task? I didn't see any accommodation for this in the Worker code, but I also didn't see any error checking.

Reply:
@andre-merzky says:

Function tasks (probably) ignore *named_env*. **We should not specify it.** (same for `pre_exec`)